install.hs

```diff
@@ -198,36 +193,30 @@ cabalInstallHie versionNumber = do
             (localBin </> "hie-" ++ dropExtension versionNumber <.> exe)
 
 cabalBuildDoc :: Action ()
-cabalBuildDoc = generateHoogleDatabase $ do 
-  localBin <- getLocalBin
-  execCabal_ ["new-install", "--symlink-bindir=" ++ localBin, "hoogle"]
+cabalBuildDoc = do
+  execCabal_ ["new-build", "hoogle", "generate"]
   execCabal_ ["new-exec", "hoogle", "generate"]
 
-generateHoogleDatabase :: Action () -> Action ()
-generateHoogleDatabase installIfNecessary = do
-  mayHoogle <- liftIO $ findExecutable "hoogle"
-  case mayHoogle of
-    Nothing -> installIfNecessary
-    Just hoogle -> command_ [] "hoogle" ["generate"]
-
-
 cabalTest :: VersionNumber -> Action ()
 cabalTest versionNumber = do
   configureCabal versionNumber
   execCabal_ ["new-test"]
 
-installCabal :: GhcPath -> Action ()
-installCabal ghc = do
-  execStack_ ["install", "--stack-yaml=shake.yaml", "cabal-install"]
+installCabal :: Action ()
+installCabal = do
+  -- install `cabal-install` if not already installed
+  unlessM (existsExecutable "cabal") $ do
+    execStack_ ["install", "--stack-yaml=shake.yaml", "cabal-install"]
   execCabal_ ["update"]
+  ghc <- getStackGhcPath mostRecentHieVersion
```

Comment:
This would require a third Stack snapshot, maybe prefer `shake.yaml`  or use the same resolver as the script?

Reply:
This is indeed an issue. Maybe `Cabal` needs to be installed using all `ghc`-versions? I suppose that it does not really have any benefit to install `Cabal` for a different `ghc` than is actually being used by `hie` (if only one `hie` version is required). 

@alanz do you know if my hypothesis about installing `Cabal` a different `ghc` is true?

Reply:
now uses the same ghc as the build process. If this actually helps with anything is another question...